test/functional/lua/utility_functions_spec.lua

```diff
@@ -106,3 +108,96 @@ describe('vim.stricmp', function()
     eq(1, funcs.luaeval('vim.stricmp("\\0C\\0", "\\0B\\0")'))
   end)
 end)
+
+describe("vim.split", function()
+  it("works", function()
+    local split = function(str, sep)
+      return funcs.luaeval('vim.split(_A.s, _A.e)', { s = str, e = sep })
```

Comment:
Use `meths.execute_lua` to avoid unnecessary indirection through vimL, also args handling is simpler just `meths.execute_lua('return vim.split(...)', {str, sep})`